packages/alfa-rules/src/common/predicate/is-rendered.ts

```diff
@@ -6,34 +7,41 @@ import { Style } from "@siteimprove/alfa-style";
 
 const { isElement, hasName } = Element;
 
+const cache = Cache.empty<[Device, Context, Node], boolean>();
```

Comment:
```suggestion
const cache = Cache.empty<Device, Cache<Context, Cache<Node, boolean>>>();
```

The `Cache` class uses reference equality, not structural equality, in order to be able to only hold weak references to object types.

Reply:
Yeah, I actually thought so yesterday night 🙈